app/src/main/java/org/coepi/android/ui/alerts/AlertsAdapter.kt

```diff
@@ -32,7 +35,19 @@ class AlertsAdapter(
         ViewHolder(parent)
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        holder.bind(getItem(position), onAlertClick)
+        val currentItem = getItem(position)
+        var previousItem: AlertViewData? = null
+        var monthHeaderVisible = true
+
+        if (position > 0) {
+            previousItem = getItem(position - 1)
+        }
+
+        previousItem?.let {
+            monthHeaderVisible = currentItem.contactTimeMonth != previousItem.contactTimeMonth
```

Comment:
This should be mixed element list (header/normal item), prepared in the view model. This kind of logic shouldn't be in the UI.

Reply:
@i-schuetz are you suggesting that I remove the header xml item out of the item_alert xml file and separate it from that view entirely? I'm trying to think of how I can separate it from the item_alert layout, but since we show these headers in the views inside the recyclerview I'm not sure how to handle it outside of the adapter

Reply:
There are 2 ways to manage this: Having the headers in the alert cells or as a separate items. To me it seems more natural as a separate item. We've other recycler views with mixed items which you can orient with.

If there was a reason to have them inside the alerts (say, if we wanted to show an overlay that spans across the header and the alerts), you _still_ would write the logic to show/hide the header in the view model, not here. The alerts view data would have a field "header" or similar.

Reply:
@i-schuetz I saw what you were talking about where I could separate them out and have a type "Item" and type "Header" for the adapter. I like that approach better, but since I have already been working on another branch for the swiping functionality I went with the faster approach. Thanks for approving; I can always go back and refactor this so that it is cleaner!